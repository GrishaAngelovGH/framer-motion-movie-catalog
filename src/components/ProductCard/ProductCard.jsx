import { motion } from "framer-motion"

const ProductCard = ({ image, title, content }) => (
  <motion.div whileHover={{ scale: 1.1 }} className="bg-white p-2 min-h-[400px] rounded-md w-2/3 text-center">
    <img src={image} className="inline-block w-full rounded-md" />
    <p className="m-5 text-xl">{title}</p>
    <p className="text-gray-500">{content}</p>
  </motion.div>
)

export default ProductCard