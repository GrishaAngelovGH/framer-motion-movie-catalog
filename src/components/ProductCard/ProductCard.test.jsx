import { render } from "@testing-library/react"

import ProductCard from "./ProductCard"

test("should render ProductCard component", () => {
  const view = render(
    <ProductCard
      image="image-1.jpg"
      title="Product Title"
      content="Product Content"
    />
  )

  expect(view).toMatchSnapshot()
})