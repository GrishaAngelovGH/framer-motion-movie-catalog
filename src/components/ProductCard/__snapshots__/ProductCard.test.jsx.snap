// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render ProductCard component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="min-w-[300px] bg-white p-2 min-h-[400px] rounded-md w-2/3 text-center"
      >
        <img
          class="inline-block w-full rounded-md"
          src="image-1.jpg"
        />
        <p
          class="m-5 text-xl"
        >
          Product Title
        </p>
        <p
          class="text-gray-500"
        >
          Product Content
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="min-w-[300px] bg-white p-2 min-h-[400px] rounded-md w-2/3 text-center"
    >
      <img
        class="inline-block w-full rounded-md"
        src="image-1.jpg"
      />
      <p
        class="m-5 text-xl"
      >
        Product Title
      </p>
      <p
        class="text-gray-500"
      >
        Product Content
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
